### ISSUE TITLE
schematic(add): does not respect workspace's defaultProject 

### ISSUE THREAD

**Issue by Splaktar** [angular/components#16743]
#### Reproduction

Steps to reproduce:
1. `ng new tmp` and respond to prompts
1. `cd tmp/`
1. `ng g app web-app` and respond to prompts
1. Edit workspace (`angular.json`) to change `"defaultProject": "tmp"` to `"defaultProject": "web-app"`
 1. `ng add @angular/material` and respond to prompts
 
#### Expected Behavior
[The `ng add` docs](https://angular.io/cli/add) state
> Adds the npm package for a published library to your workspace, and configures your default app project to use that library,

> The default app project is the value of `defaultProject` in `angular.json`.

Thus it is expected to add Angular Material to the default project, `web-app`.

#### Actual Behavior
```bash
Skipping installation: Package already installed
? Choose a prebuilt theme name, or "custom" for a custom theme: Custom
? Set up HammerJS for gesture recognition? Yes
? Set up browser animations for Angular Material? Yes
UPDATE package.json (1473 bytes)
UPDATE src/main.ts (391 bytes)
UPDATE src/app/app.module.ts (423 bytes)
UPDATE src/styles.scss (1418 bytes)
UPDATE src/index.html (484 bytes)
```
As you can see, it configured the initial app project in the workspace (`"tmp"`) for Angular Material instead of the default project (`"web-app"`).

#### Environment
```bash
Angular CLI: 8.2.1
Node: 10.15.3
OS: darwin x64
Angular: 8.2.1
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.802.1
@angular-devkit/build-angular      0.802.1
@angular-devkit/build-ng-packagr   0.802.1
@angular-devkit/build-optimizer    0.802.1
@angular-devkit/build-webpack      0.802.1
@angular-devkit/core               8.2.1
@angular-devkit/schematics         8.2.1
@angular/cdk                       8.1.2
@angular/material                  8.1.2
@ngtools/webpack                   8.2.1
@schematics/angular                8.2.1
@schematics/update                 0.802.1
ng-packagr                         5.4.3
rxjs                               6.4.0
typescript                         3.5.3
webpack                            4.38.0
```

**Comment by devversion** [comment]
@Splaktar I investigated this and it looks like we are not doing anything magical for retrieving the project name.

We leverage the CLI's `projectName` provider which apparently determines the current project based on path and falls back to the `defaultProjectName` property if nothing could be found by path.

Problem is that the `ng add` command is executed in the workspace root which is also the project root of the `tmp` project --> so by path the current project is `tmp`. The issue would not appear if both projects (`tmp` and `web-app`) are under a `projects/` folder.

So I think this is actually a CLI issue and it might be best to move that over? thoughts?

**Comment by Splaktar** [comment]
Yeah, that code doesn't seem to match up with docs that I linked above.

@alan-agius4 is this a CLI docs issue (and should be transferred to angular/angular) or a CLI issue (and go to angular/angular-cli)?

**Comment by alan-agius4** [comment]
@Splaktar, what @devversion explained is correct and we use the current working directory to determine the project hence what you are seeing is the expected behaviour.

The docs highlighted above needs to be fixed, these docs are in the CLI repo, so probably makes sense to transfer this to the CLI. 

_I don’t seem to have permissions to transfer issues from the components repo._
